darwin/client.py

```diff
@@ -150,7 +150,7 @@ def get_remote_dataset(self, dataset_identifier: Union[str, DatasetIdentifier])
                 client=self,
             )
         if not matching_datasets:
-            raise NotFound(parsed_dataset_identifier)
+            raise NotFound(str(parsed_dataset_identifier))
```

Comment:
MyPy rightly complained about this line, as `NotFound` accepts a `str`-typed parameter, and `parsed_dataset_identifier` is of type `DatasetIdentifier`